src/lib/paginator/paginator.ts

```diff
@@ -100,6 +100,18 @@ export class MatPaginator extends _MatPaginatorBase implements OnInit, OnDestroy
   get length(): number { return this._length; }
   set length(value: number) {
     this._length = coerceNumberProperty(value);
+
+    const maxPageIndex = Math.max(this.getNumberOfPages() - 1, 0);
+    const currentPageIndex = this._pageIndex;
+
+    if (currentPageIndex > maxPageIndex && this.initialized) {
+      // Needs to happen on the next tick, in order to avoid "changed after checked" errors.
```

Comment:
Why does this cause a changed after checked error?

Reply:
It's been a while so I don't remember the _exact_ reason. I think it had to do with the page event being emitted at the wrong time the which would cause the table to be re-rendered after the changes were detected.